RecoPixelVertexing/PixelTriplets/plugins/PixelQuadrupletEDProducer.cc

```diff
@@ -0,0 +1,128 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Utilities/interface/RunningAverage.h"
+
+#include "RecoTracker/TkHitPairs/interface/RegionsSeedingHitSets.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/OrderedHitSeeds.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/IntermediateHitTriplets.h"
+
+#include "PixelQuadrupletGenerator.h"
+#include "LayerQuadruplets.h"
+
+class PixelQuadrupletEDProducer: public edm::stream::EDProducer<> {
+public:
+  PixelQuadrupletEDProducer(const edm::ParameterSet& iConfig);
+  ~PixelQuadrupletEDProducer() = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  virtual void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+
+private:
+  edm::EDGetTokenT<IntermediateHitTriplets> tripletToken_;
+
+  edm::RunningAverage localRA_;
+
+  PixelQuadrupletGenerator generator_;
+};
+
+PixelQuadrupletEDProducer::PixelQuadrupletEDProducer(const edm::ParameterSet& iConfig):
+  tripletToken_(consumes<IntermediateHitTriplets>(iConfig.getParameter<edm::InputTag>("triplets"))),
+  generator_(iConfig, consumesCollector())
+{
+  produces<RegionsSeedingHitSets>();
+}
+
+void PixelQuadrupletEDProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("triplets", edm::InputTag("hitTripletEDProducer"));
+  PixelQuadrupletGenerator::fillDescriptions(desc);
+
+  descriptions.add("pixelQuadrupletEDProducer", desc);
+}
+
+void PixelQuadrupletEDProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<IntermediateHitTriplets> htriplets;
+  iEvent.getByToken(tripletToken_, htriplets);
+  const auto& regionTriplets = *htriplets;
+
+  const SeedingLayerSetsHits& seedingLayerHits = regionTriplets.seedingLayerHits();
+  if(seedingLayerHits.numberOfLayersInSet() < 4) {
+    throw cms::Exception("Configuration") << "PixelQuadrupletEDProducer expects SeedingLayerSetsHits::numberOfLayersInSet() to be >= 4, got " << seedingLayerHits.numberOfLayersInSet();
```

Comment:
similar case, looks more like a LogicError than a Configuration given how it's detected